src/findings_data_import.py

```diff
@@ -49,12 +49,292 @@ def move_processed_file(s3_client, bucket, folder, filename):
         logging.error("Error: %s", delete_error)
 
 
-def skip_record(index, file, format_string="", *args):
-    """Print a standard message when a record must be skipped."""
-    log_format = 'Skipping record %d of "%s"' + (
-        f": {format_string}" if format_string else ""
+def get_field_map(s3_client=None, s3_bucket=None, field_map=None):
+    """Read a JSON field map object from a s3 bucket, and return the field map dict.
+
+    Parameters:
+    -----------
+    s3_client : S3.Client
+        The AWS S3 Client to retrieve the data file with.
+
+    s3_bucket : str
+        The AWS S3 bucket containing the data file.
+
+    field_map : str
+        The S3 key for the JSON file containing a map of incoming field names
+        to what they should be for the database.
+
+    Returns
+    -------
+    dict : The field map dict object
+    """
+    try:
+        # Log what/where up front so the subsequent messages make more sense
+        logging.info(
+            "Attempting to read Configuration data from %s in %s", field_map, s3_bucket
+        )
+        # Fetch object for the field_map JSON
+        field_map_object = s3_client.get_object(Bucket=s3_bucket, Key=field_map)
+        # Load field_map JSONs
+        field_map_dict = json.loads(
+            field_map_object.get("Body", "{}").read().decode("utf-8")
+        )
+        logging.info("Configuration data loaded from %s", field_map)
+        logging.debug("Configuration data: %s", field_map_dict)
+
+        return field_map_dict
+    except ClientError as client_err:
+        raise Exception(
+            f"Unable to download the field map data {field_map} from {s3_bucket}",
+            client_err,
+        )
+    except json.JSONDecodeError as json_err:
+        raise Exception(
+            "Unable to decode field map data, does not appear to be valid JSON.",
+            json_err,
+        )
+
+
+def setup_database_connection(
+    db_hostname=None,
+    db_port=None,
+    ssm_db_name=None,
+    ssm_db_user=None,
+    ssm_db_password=None,
+):
+    """Set up a mongo db connection based on the supplied host/port and SSM key values.
+
+    Parameters
+    ----------
+    db_hostname : str
+        The hostname that has the database to store the data in.
+
+    db_port : str
+        The port that the database server is listening on. [default: 27017]
```

Comment:
Why wouldn't this be an `int`?

Reply:
Because, inexplicably, it already is in the current production branch and I didn't want to make any more waves than already accomplished

The current code uses a hard-coded port string explicitly, and documents it as a string. And really, since we're working with python the typing is more of a suggestion than anything else.

https://github.com/cisagov/findings-data-import-lambda/blob/2c3ed77281388897a149e60c278a42652a40c367/src/findings_data_import.py#L60-L87

Reply:
Fair enough.

Reply:
> And really, since we're working with python the typing is more of a suggestion than anything else.

That is kind of true, but it's always good to document any constraints you are expecting on input variables.  In this case `db_port` is pretty clearly expected to be an `int` (or at least a `str` that can be parsed into an `int`).

Reply:
I don't disagree at all, it should be an int unless there is an explicit reason. I mostly just want to limit the scope of this change where possible, but can edit it here in this PR as well if need be.